src/profile-logic/tracks.js

```diff
@@ -485,10 +485,14 @@ export function getVisibleThreads(
         hiddenTracksByPid.get(pid),
         'Hidden tracks were expected to exists for the given pid.'
       );
-      for (const track of tracks) {
-        if (track.type === 'thread') {
-          const { threadIndex } = track;
-          if (!hiddenTracks.has(threadIndex)) {
+      for (
+        let localTrackIndex = 0;
+        localTrackIndex < tracks.length;
+        localTrackIndex++
+      ) {
+        if (tracks[localTrackIndex].type === 'thread') {
+          const { threadIndex } = tracks[localTrackIndex];
```

Comment:
nit: can you please set the `track` in a local variable at the start of the loop, to make it easier to read?
`const track = tracks[localTrackIndex];` right at the start of the loop, and then use it in the rest of the code.